### PR TITLE
Gsoap fix

### PR DESCRIPTION
**Pull request by KristianJerpetjon**
Specify library name and version:  gsoap/all

Avoids building gsoap tools when cross compiling.. (not doable as it builds a host tool to build another host tool)
exports the gsoap plugin structure often used by projects

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — COMMENTED** [review]
Since you are adding the C lib, we should probably test that it is generated correctly.

I also noticed that the pkg_conf components are not correctly modeling the upstream... since you are adding a second one it might be worth noting they are wrong...

**Comment by prince-chrismc** [comment]
Might be worth adding 2.8.108 as well

File: recipes/gsoap/all/conanfile.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def _configure_cmake(self):
             return self._cmake
         self._cmake = CMake(self)
         self._cmake.definitions["GSOAP_PATH"] = self._source_subfolder
-        self._cmake.definitions["BUILD_TOOLS"] = True
+        self._cmake.definitions["BUILD_TOOLS"] = not tools.cross_building(self.settings)
```
Comment (madebr) [inline recipes/gsoap/all/conanfile.py:66]: I'm not a fan of creating different packages (same hash, different content) depending on the build system.
I would add a `tools` option and fail in `configure` when cross building and this option is set.

**Review by uilianries — APPROVED** [review]

File: recipes/gsoap/all/conanfile.py (hunk 1 above)
Comment (danimtb) [reply recipes/gsoap/all/conanfile.py:66]: Agree with this comment. Every change in the resulting package should be a consequence o changing an option or a setting value. In this case, why not leaving the build tools def to `True`?

Comment (KristianJerpetjon) [reply recipes/gsoap/all/conanfile.py:66]: Build tools true when cross compiling fails on itself due to recursion.

Comment (madebr) [reply recipes/gsoap/all/conanfile.py:66]: I just noticed that we don't use the gsoap autotools build system at all.
We apparently have forked our own cmake system.
See https://github.com/conan-io/conan-center-index/pull/1699 for the pr adding gsoap. I don't know what problems the autotools build sytem has? (@ericLemanissier  @madduci)

Anyhow:
- the cmake system has not been used to cross build between Windows and Linux due to it wanting to use winflexbison on mingw@Linux.
- the test_package does not work when the tools aren't built

I've created https://github.com/conan-io/conan-center-index/pull/3615 which addresses some of these issues.

@KristianJerpetjon 
Can you try it? I've created a new `tools` option.

Comment (ericLemanissier) [reply recipes/gsoap/all/conanfile.py:66]: I have no idea why upstream autotools system is not used

**Pull request closed without merging**